openqabot/types/baseconf.py

```diff
@@ -27,8 +27,10 @@ def __call__(
     def normalize_repos(config):
         pass
 
-    def filter_embargoed(self) -> bool:
-        return any(k.startswith("PUBLIC") for k in self.settings.keys())
+    def filter_embargoed(self, flavor: str) -> bool:
+        return any(k.startswith("PUBLIC") for k in self.settings) or any(
+            flavor.startswith(s) for s in ("Azure", "EC2", "GCE")
```

Comment:
Is using of  `startswith`  too restrictive?

It will works as expected but I'm not sure there's any "official rule" that say that name of the cloud provider has to be used at the beginning of the `FLAVOR` name.

More deeply in the PC test lib there's a more coded convention based on `PUBLIC_CLOUD_PROVIDER` and not on `FLAVOR` 

There's a PC lib setting named `PUBLIC_CLOUD_PROVIDER` that has a coded naming convention: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/2e0b43a6f0cea5d2e8856e2a35d8bb23bb625325/lib/publiccloud/basetest.pm#L33

Then, at least for the QE-SAP jobs the connection between FLAVOR and PUBLIC_CLOUD_PROVIDER happens at JobGroup level (have a look for the setting name in JobGroup file `openQA/yaml_job_groups/maintenance/maintenance_sle15_sp5_sap.yml` in repo `openqa_ha_sap`

Reply:
the problem which @mimi1vx  was trying to address here that currently https://gitlab.suse.de/qa-maintenance/metadata does not have `PUBLIC_CLOUD_PROVIDER` for SAP related products . And if you will add it there than you won't need second part at all . Because `any(k.startswith("PUBLIC") for k in self.settings)` would already return true for `PUBLIC_CLOUD_PROVIDER` variable no matter what value it has